examples/DotNetCore/ConsoleApplicationWithMap/ConsoleApplicationWithMap/Program.cs

```diff
@@ -0,0 +1,158 @@
+
+            // Finish on key press
+            Console.ReadKey();
+            cts.Cancel();
+        }
+
+        private static async Task<MyBlobContent?> ReadBlobContentAsync(Uri blobUri)
+        {
+            var blobClient = new BlobClient(blobUri, new DefaultAzureCredential());
+
+            // Read blob content
+            var contentStream = new MemoryStream();
+            await blobClient.DownloadToAsync(contentStream);
+
+            // Go to the beginning of the content stream
+            contentStream.Position = 0;
+
+            // Convert content to json format
+            var reader = new StreamReader(contentStream);
+            var jsonReader = new JsonTextReader(reader);
+
+            var serializer = new JsonSerializer();
+
+            // Save the string values in the JSON array "Data" to MyBlobContent.Data
+            return serializer.Deserialize<MyBlobContent>(jsonReader);
+        }
+
+        private static void Configure()
+        {
+            var builder = new ConfigurationBuilder();
+
+            // Load a subset of the application's configuration from a json file and environment variables
+            builder.AddJsonFile("appsettings.json")
+                   .AddEnvironmentVariables();
+
+            IConfiguration configuration = builder.Build();
+
+            if (string.IsNullOrEmpty(configuration["ConnectionString"]))
+            {
+                Console.WriteLine("Connection string not found.");
+                Console.WriteLine("Please set the 'ConnectionString' environment variable to a valid Azure App Configuration connection string and re-run this example.");
+                return;
+            }
+
+            // Augment the configuration builder with Azure App Configuration
+            // Pull the connection string from an environment variable
+            builder.AddAzureAppConfiguration(options =>
+            {
+                options.Connect(configuration["ConnectionString"])
+                        .Select("*")
+                        .ConfigureRefresh(refresh =>
+                        {
+                            // key: BlobUri
+                            // value: https://{account_name}.blob.core.windows.net/{container_name}/{blob_name}
+                            refresh.Register("BlobUri")
+                                    .SetCacheExpiration(TimeSpan.FromSeconds(10));
+                        })
+                        .Map(async (setting) =>
+                        {
+                            if (setting.ContentType.Equals("application/storage.blob"))
+                            {
+                                MyBlobContent? blobContent = await ReadBlobContentAsync(new Uri(setting.Value));
+                                string newSettingValue = "";
+
+                                if (blobContent != null)
+                                {
+                                    newSettingValue = $"[{string.Join(", ", blobContent.Data)}]";
```

Comment:
What is the motivation behind converting blob data into a value that looks like a list representation?

Reply:
The example from the customer issue converted the blob with text in json format to a list, so I was copying that assuming it was significant somehow. I can also just directly copy the file as a string to the setting value.

Reply:
If we want to show the full binding example, we could change the content type of mapped setting to application/json. That way, our JSON adapter will process the mapped list value and customers can bind the final IConfiguration value directly to a List object.

Reply:
That sounds like a good idea, I'll test that out

Reply:
Seems to work, I've pushed some changes with revision but I think I'm still unsure on how the display should look